clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -2774,9 +2774,13 @@ def err_constexpr_tag : Error<
 def err_constexpr_dtor : Error<
   "destructor cannot be declared %sub{select_constexpr_spec_kind}0">;
 def err_constexpr_dtor_subobject : Error<
-  "destructor cannot be declared %sub{select_constexpr_spec_kind}0 because "
+  "destructor cannot be declared %sub{select_constexpr_spec_kind}0 before C++23 because "
   "%select{data member %2|base class %3}1 does not have a "
   "constexpr destructor">;
+def warn_cxx20_compat_constexpr_dtor_subobject : Warning<
```

Comment:
We usually only issue the compat warnings in cases where we expose it as an extension in earlier modes, so I think this warning can go away.

Reply:
Done too.